pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -293,16 +297,62 @@ func (c *Controller) syncOwnedProvisionRequest(
 				if !psFound || !psaFound {
 					return nil, errInconsistentPodSetAssignments
 				}
-				req.Spec.PodSets = append(req.Spec.PodSets, autoscaling.PodSet{
+
+				pt := &corev1.PodTemplate{}
+				ptKey := types.NamespacedName{
+					Namespace: wl.Namespace,
+					Name:      getProvisioningRequestPodTemplateName(requestName, psName),
+				}
+
+				podSet := autoscaling.PodSet{
 					PodTemplateRef: autoscaling.Reference{
-						Name: getProvisioningRequestPodTemplateName(requestName, psName),
+						Name: ptKey.Name,
 					},
 					Count: ptr.Deref(psa.Count, ps.Count),
-				})
-			}
+				}
 
-			if err := ctrl.SetControllerReference(wl, req, c.client.Scheme()); err != nil {
-				return nil, err
+				err := c.client.Get(ctx, ptKey, pt)
+				if client.IgnoreNotFound(err) != nil {
+					return nil, err
+				}
+				if err != nil {
+					// it's a not found, so create it
+					newPt := &corev1.PodTemplate{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      ptKey.Name,
+							Namespace: ptKey.Namespace,
+							Labels: map[string]string{
+								constants.ManagedByKueueLabel: "true",
+							},
+						},
+						Template: ps.Template,
+					}
+
+					// apply the admission node selectors to the Template
+					psi, err := podset.FromAssignment(ctx, c.client, psaMap[psName], podSet.Count)
+					if err != nil {
+						return nil, err
+					}
+
+					err = podset.Merge(&newPt.Template.ObjectMeta, &newPt.Template.Spec, psi)
+					if err != nil {
+						return nil, err
+					}
+
+					// copy limits to requests if needed
+					workload.UseLimitsAsMissingRequestsInPod(&newPt.Template.Spec)
+
+					if err = c.client.Create(ctx, newPt); err != nil {
+						msg := fmt.Sprintf("Error creating PodTemplate %q: %v", newPt.Name, err)
+						ac.Message = api.TruncateConditionMessage(msg)
+						workload.SetAdmissionCheckState(&wl.Status.AdmissionChecks, *ac, c.clock)
+
+						c.record.Eventf(wl, corev1.EventTypeWarning, "FailedCreate", api.TruncateEventMessage(msg))
+						return nil, err
+					}
```

Comment:
Could we make a separate function for PodTemplate creation?

Reply:
Done